iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber_uds.inl

```diff
@@ -0,0 +1,146 @@
+                          .value()))
+{
+}
+
+template <typename T>
+inline void EventSubscriberUds<T>::Subscribe(std::size_t) noexcept
+{
+    // Subscribe not supported with UDS
+}
+
+template <typename T>
+void EventSubscriberUds<T>::Unsubscribe() noexcept
+{
+    // Unsubscribe not supported with UDS
+}
+
+template <typename T>
+template <typename Callable>
+core::Result<size_t> inline EventSubscriberUds<T>::GetNewSamples(Callable&& callable,
+                                                                 size_t maxNumberOfSamples) noexcept
+{
+    IOX_DISCARD_RESULT(maxNumberOfSamples);
+
+    core::Result<size_t> numberOfSamples{1};
+
+    auto samplePtr = std::make_unique<SampleType>();
+    SampleType& sample = *samplePtr;
+
+    std::string tempBuffer;
+
+    // Receive the first (up to) 4095 Bytes
+    m_uds.receive().and_then([&](auto& msg) { tempBuffer.append(msg); }).or_else([](auto&) {
+        std::cerr << "Error occurred while receiving UNIX domain socket!" << std::endl;
+    });
+
+    if (tempBuffer.size() == 0)
+    {
+        return 0;
+    }
+
+    // Deserialize the counter
+    sample.counter =
+        (static_cast<uint32_t>(tempBuffer[3]) & 0xFF) << 24 | (static_cast<uint32_t>(tempBuffer[2]) & 0xFF) << 16
+        | (static_cast<uint32_t>(tempBuffer[1]) & 0xFF) << 8 | (static_cast<uint32_t>(tempBuffer[0]) & 0xFF);
+
+    // Deserialize the timestamp
+    uint64_t sendTimestamp =
+        (static_cast<uint64_t>(tempBuffer[11]) & 0xFF) << 56 | (static_cast<uint64_t>(tempBuffer[10]) & 0xFF) << 48
+        | (static_cast<uint64_t>(tempBuffer[9]) & 0xFF) << 40 | (static_cast<uint64_t>(tempBuffer[8]) & 0xFF) << 32
+        | (static_cast<uint64_t>(tempBuffer[7]) & 0xFF) << 24 | (static_cast<uint64_t>(tempBuffer[6]) & 0xFF) << 16
+        | (static_cast<uint64_t>(tempBuffer[5]) & 0xFF) << 8 | (static_cast<uint64_t>(tempBuffer[4]) & 0xFF);
+    int64_t castedSendTimestamp = static_cast<int64_t>(sendTimestamp);
+    sample.sendTimestamp = std::chrono::time_point<std::chrono::steady_clock>(
+        std::chrono::duration<int64_t, std::nano>(castedSendTimestamp));
+
+    // Deserialize subPackets
+    uint32_t subPackets =
+        (static_cast<uint32_t>(tempBuffer[15]) & 0xFF) << 24 | (static_cast<uint32_t>(tempBuffer[14]) & 0xFF) << 16
+        | (static_cast<uint32_t>(tempBuffer[13]) & 0xFF) << 8 | (static_cast<uint32_t>(tempBuffer[12]) & 0xFF);
+    sample.subPackets = subPackets;
+
+    // If more than 4095 Bytes were send in consecutive messages, receive them now
+    if (subPackets > 1)
+    {
+        for (uint32_t i = 0U; i < subPackets - 1; ++i)
+        {
+            m_uds.receive().and_then([&](auto& msg) { tempBuffer.append(msg); }).or_else([](auto&) {
+                std::cerr << "Error occurred while receiving UNIX domain socket!" << std::endl;
+            });
+        }
+    }
+
+    // Complete fragmented message was received, now we call the user-defined callable
+    callable(samplePtr);
```

Comment:
The actual content of the message is still in the `tempBuffer` I think, so calling it is very dangerous. Unless I missed something and to be fair we need to copy the data into the sample first. Ideally we would copy the bytes there directly when we receive them, which should be possible if we can allocate the sample once we know the size from the first packet.

Any comparison with iceoryx that does not do this is already artificially unfair to UDS.

Reply:
Yeah, this really comes down to that it is only used in a specific way for benchmarking. But if we ever try to call a callable that actually wants to use the data bad things will happen. OK with me since it is in the UDS part only.